### PR TITLE
allow description or extended notes on policies

### LINKED ISSUE #1814: Allow description or extended notes on policies
There can be a lot of policies. And sometimes you are wondering, why there is a policy.

We might allow to add "notes" or "description" to policies. This would be something like an internal documentation:

"This policy adds the user-details to the response. This is needed for the connected benno mailarchive. Otherwise login will not be possible. Do not delete this policy!"

I am not quite sure, if we should do such things, because it is somehow also a documentation. And the system should not be documented within the system itself.

#2146

### PR DESCRIPTION
**Pull request by jona-samuel**
closes #1814

### CODE REVIEW

File: privacyidea/models.py [hunk 1]
```diff
@@ -1514,6 +1514,36 @@ def cleanup_challenges():
     Challenge.query.filter(Challenge.expiration < c_now).delete()
     db.session.commit()
 
+
+#-----------------------------------------------------------------------------
+#
+#DESCRIPTION
```
Comment (nilsbehlen) [inline privacyidea/models.py:1520]: space after #

File: privacyidea/static/components/config/views/config.policies.details.html [hunk 2]
```diff
@@ -220,7 +220,7 @@ <h3 translate>Condition</h3>
                            ng-model="params.user"
                            placeholder="userA, userB"/>
                 </div>
-            </div>
+            </div>
```
Comment (nilsbehlen) [inline privacyidea/static/components/config/views/config.policies.details.html:223]: remove space after >

**Review by nilsbehlen — CHANGES_REQUESTED** [review]
just formatting stuff, please also fix the pep bot comments

File: privacyidea/static/components/config/views/config.policies.details.html (hunk 2 above)
Comment (jona-samuel) [reply privacyidea/static/components/config/views/config.policies.details.html]: [659ce1b](https://github.com/privacyidea/privacyidea/pull/3870/commits/659ce1b93e36bb277dd27ed785d09e9ebc8fb1c1)

File: privacyidea/models.py (hunk 1 above)
Comment (jona-samuel) [reply privacyidea/models.py]: [fb686f7](https://github.com/privacyidea/privacyidea/pull/3870/commits/fb686f7ff439c92ccfe678238bb5b494b71cddf3)

File: doc/policies/index.rst [hunk 3]
```diff
@@ -52,6 +52,9 @@ Each policy can contain the following attributes:
      On a library level or during migration scripts policies with
      other characters could be created.
 
+**description**
+    Use this to describe your policy in more detail
```
[the remark this answers is not in the document]
Comment (jona-samuel) [reply doc/policies/index.rst]: [fb686f7](https://github.com/privacyidea/privacyidea/pull/3870/commits/fb686f7ff439c92ccfe678238bb5b494b71cddf3)

**Review by nilsbehlen — APPROVED** [review]

File: migrations/versions/5741e5dac477_.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            create_seq(seq)
+        except Exception as _e:
+            pass
+        op.create_table('description',
+        sa.Column('id', sa.Integer(), seq, nullable=False),
+        sa.Column('name', sa.Unicode(length=64), nullable=False),
+        sa.Column('object_type', sa.Unicode(length=64), nullable=False),
+        sa.Column('last_update', sa.DateTime),
+        sa.Column('description', sa.UnicodeText()),
+        sa.ForeignKeyConstraint(['object_id'], ['policy.id'], ),
```
Comment (plettich) [inline migrations/versions/5741e5dac477_.py:41]: This doesn't work, the column `object_id` needs to be added as well.

**Review by plettich — CHANGES_REQUESTED** [review]

File: migrations/versions/5741e5dac477_.py (hunk 4 above)
Comment (jona-samuel) [reply migrations/versions/5741e5dac477_.py:42]: [4212903](https://github.com/privacyidea/privacyidea/pull/3870/commits/421290318fc25cb5c6d73d1fff7acc8224e15712)

File: migrations/versions/5741e5dac477_.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
+"""empty message
```
[the remark this answers is not in the document]
Comment (jona-samuel) [reply migrations/versions/5741e5dac477_.py]: [4212903](https://github.com/privacyidea/privacyidea/pull/3870/commits/421290318fc25cb5c6d73d1fff7acc8224e15712)

**Review by plettich — APPROVED** [review]

**Pull request merged**